spec/controllers/deliverers_controller_spec.rb

```diff
@@ -44,6 +44,56 @@
     end
   end
 
+  # Test index action with params
+  describe 'get #index with params' do
+    Given!(:deliverers) { FactoryGirl.create_list(:deliverer, 15) }
+    Given!(:deliverer_to_search) do
+      FactoryGirl.create(:deliverer,
+                         name: 'Bobby Lee',
+                         phone: 123,
+                         vehicle: 2,
+                         active: true)
+    end
+
+    context 'when query is blank' do
+      When do
+        get :index, params: {
+          query: {
+            name_cont: '',
+            phone_cont: '',
+            vehicle_eq: '',
+            active_eq: ''
+          }
+        }, format: 'js', xhr: true
+      end
+
+      Then { expect(assigns(:deliverers).length).to eq 16 }
+    end
+
+    context 'when query contains name partial' do
+      When { get :index, params: { query: { name_cont: 'le' } } }
+
+      Then { expect(assigns(:deliverers).length).to eq 1 }
```

Comment:
we should also check whether the result is the correct deliverer

Reply:
Fixed, the following test for filter form results now uses object equivalence instead.